Sources/Transport/URI.swift

```diff
@@ -12,15 +12,15 @@ public struct URI {
   private var components: URLComponents
 
   /// Creates a URI from the provided path, query string.
-  public init(path: String = "/", query: String? = nil) {
+  public init(path: String = "/", queryItems: [URLQueryItem]? = nil) {
```

Comment:
This unfortunately could break user code. Can you keep this one as the primary initializer and then add an additional initializer that takes the `String`? 

```swift
public init(path: String, query: String? = nil) {
  let queryItems = URLComponents(string: url.absoluteString)?.queryItems
  self.init(path: components.path, queryItems: queryItems)
}
```

That way we don't have to make any changes to `WebSocketClient`

Reply:
OK